workflow/controller/operator.go

```diff
@@ -975,7 +975,8 @@ func (woc *wfOperationCtx) podReconciliation(ctx context.Context) error {
 
 			// If the node is pending and the pod does not exist, it could be the case that we want to try to submit it
 			// again instead of marking it as an error. Check if that's the case.
-			if node.Pending() {
+			// Node will be in pending state without Pod create if Node is waiting for Synchronize lock
+			if node.Pending() && !node.IsPodCreated() {
```

Comment:
There is one valid scenario where node status will be `pending` without pod creation. It is the template that is waiting for sync lock to acquire.